velox/functions/prestosql/aggregates/DecimalAggregate.h

```diff
@@ -240,12 +245,14 @@ class DecimalAggregate : public exec::Aggregate {
         if (decodedPartial_.isNullAt(i)) {
           return;
         }
+        clearNull(group);
         auto decodedIndex = decodedPartial_.index(i);
         auto serializedAccumulator =
             intermediateFlatVector->valueAt(decodedIndex);
         mergeAccumulators(group, serializedAccumulator);
       });
     } else {
+      clearNull(group);
```

Comment:
Also this one need to check `rows.hasSelections()`

Reply:
Updated.